push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/apns/ApnsUtil.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright Red Hat, Inc., and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.unifiedpush.message.sender.apns;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.security.auth.x500.X500Principal;
+import java.io.ByteArrayInputStream;
+import java.security.KeyStore;
+import java.security.cert.X509Certificate;
+import java.util.Enumeration;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class ApnsUtil {
+
+    private static final String ALGORITHM = "PKCS12";
+    private static final String TOPIC_PATTERN = ".*UID=([^,]+).*";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ApnsUtil.class);
+
+    private ApnsUtil() {
+    }
+
+    public static String readDefaultTopic(final byte[] cert, final String password) {
```

Comment:
`cert` is an array of byte representing a `keystore`, not a certificate. I think naming this `cert` is misleading.
Passwords should never be stored into Strings (strings are pooled and are immutable, so they have big chance to remain in memory for a long time)